docs/reference_guides/model_libraries/generic/costing/process_costing.rst

```diff
@@ -0,0 +1,891 @@
+    # set inputs
+    m.fs.unit.shell_inlet.flow_mol[0].fix(100)     # mol/s
+    m.fs.unit.shell_inlet.enth_mol[0].fix(3500)    # j/s
+    m.fs.unit.shell_inlet.pressure[0].fix(101325)  # Pa 
+    
+    m.fs.unit.tube_inlet.flow_mol[0].fix(100)
+    m.fs.unit.tube_inlet.enth_mol[0].fix(4000)
+    m.fs.unit.tube_inlet.pressure[0].fix(101325.0)
+    
+    m.fs.unit.area.fix(1000)  # m2
+    m.fs.unit.overall_heat_transfer_coefficient.fix(100)  # W/m2K
+    
+    m.fs.costing = SSLWCosting()
+    m.fs.H101.costing = UnitModelCostingBlock(
+        default={
+            "flowsheet_costing_block": m.fs.costing,
+            "costing_method": SSLWCostingData.cost_fired_heater,
+            "costing_method_arguments": {
+                    "material_type": HeaterMaterial.CarbonSteel,
+                    "heat_source": HeaterSource.Fuel
+            }
+        }
+    )
+
+    m.fs.unit.initialize()
+
+    opt = SolverFactory('ipopt')
+    opt.options = {'tol': 1e-6, 'max_iter': 50}
+    results = opt.solve(m, tee=True)
+
+Unit Mapping
+^^^^^^^^^^^^
+
+As a convenience for users, common unit models are mapped with appropriate costing methods if no costing method is passed:
+
+Table 2. Mapped costing method for each unit model.
+
+=========================== =========================
+Unit Model                  Costing Method
+=========================== =========================
+CSTR                        cost_vertical_vessel
+Compressor                  cost_compressor
+Flash                       cost_vertical_vessel
+Heater                      cost_fired_heater
+HeatExchanger               cost_heat_exchanger
+HeatExchangerNTU            cost_heat_exchanger
+PFR                         cost_horizontal_vessel
+PressureChanger             cost_pressure_changer
+Pump                        cost_pump
+StoichiometricReactor       cost_horizontal_vessel
+Turbine                     cost_turbine
+=========================== =========================
+
+The mapping follows unit model hierarchy (inheritance) and the framework will find the best match among available costing methods. For example, pumps will try costing with `cost_pump()` and fall back to `cost_pressure_changer()`.
+
+Heat Exchanger Cost
+^^^^^^^^^^^^^^^^^^^
+
+.. note:: IDAES does not yet support costing for heat exchangers requiring both shell and tube area (HX1D).
+
+The purchse cost is computed based on the base unit cost and three correction factors (Eq. 22.43 in Seider et al.), and is adjusted by user-defined currency units with the appropriate CE index value. The base cost is computed depending on the heat exchanger type selected by the user.
+
+The heat exchanger costing method has three arguments, hx_type = heat exchanger type, material_type = construction material, and tube_lengh = tube length.
+
+* hx_type : 
+    * 'floating_head'
+    * 'fixed_head'
+    * 'Utube'\*
+    * 'Kettle_vap'
+* material_type: 
+    * 'CarbonSteelCarbonSteel'
```

Comment:
This probably need clarification to indicate that the two material represent the shell and tube (?) along with which comes first.